cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -479,7 +475,8 @@ public void updateApp(FullHttpRequest request, HttpResponder responder,
     }
 
     try {
-      applicationLifecycleService.updateApp(appId, appRequest, createProgramTerminator());
+      applicationLifecycleService.updateApp(new ApplicationId(namespaceId, appName), appRequest,
+                                            createProgramTerminator());
```

Comment:
This probably also should  take namespace and appName. Overall ANY usage of two-parameter ApplicationId constructor should be reviewed. Probably that constructor must be removed for good

Reply:
Created this JIRA for the refactor https://cdap.atlassian.net/browse/CDAP-19937